### PR TITLE
Remove Servlet 2.5 and 3.0 Support for Remember Me and CSRF

### PR DESCRIPTION
**Pull request by dongmyo**
Remove Servlet 2.5 and 3.0 Support for Remember Me and CSRF

Fixes: gh-6263, Fixes: gh-6262

### CODE REVIEW

**Comment by jzheaux** [comment]
@dongmyo Thanks for the PR. Would you do me the favor of changing your commit title and message to include #6262 as well?

**Comment by dongmyo** [comment]
@jzheaux i changed the commit message.

**Comment by jzheaux** [comment]
Now that https://github.com/spring-projects/spring-security/pull/6276 is merged, @dongmyo, will you please do a quick rebase before we merge this one?

**Comment by dongmyo** [comment]
@jzheaux i rebased

File: web/src/main/java/org/springframework/security/web/authentication/rememberme/AbstractRememberMeServices.java [hunk 1]
```diff
@@ -424,12 +419,7 @@ protected void setCookie(String[] tokens, int maxAge, HttpServletRequest request
 			cookie.setSecure(useSecureCookie);
 		}
 
-		if (setHttpOnlyMethod != null) {
```
Comment (jzheaux) [inline web/src/main/java/org/springframework/security/web/authentication/rememberme/AbstractRememberMeServices.java]: Will you please double-check the javadoc for the `setCookie` method here? I believe it still refers to Servlet 3.0 on line 394.

Comment (dongmyo) [reply web/src/main/java/org/springframework/security/web/authentication/rememberme/AbstractRememberMeServices.java]: @jzheaux yeah, you're right. thank you.
i missed the javadoc. i'm gonna fix it.

**Comment by jzheaux** [comment]
Thanks again, @dongmyo! This is now merged into `master`.

**Pull request merged**